src/microbe_stage/organelle_components/MovementComponent.cs

```diff
@@ -141,9 +155,13 @@ private float CalculateMovementForce(CompoundBag compounds, Vector3 wantedMoveme
         SetSpeedFactor(newAnimationSpeed);
 
         if (isBacteria)
-            return Constants.FLAGELLA_BASE_FORCE * forceMagnitude;
+        {
+            return (Constants.FLAGELLA_BASE_FORCE + Constants.FLAGELLA_MAX_UPGRADE_FORCE * flagellumLength)
```

Comment:
I think it would be better to refactor out a new variable called `var baseForce = Constants.FLAGELLA_BASE_FORCE + Constants.FLAGELLA_MAX_UPGRADE_FORCE * flagellumLength` than have a really long calculation on line 163 and 164.

Reply:
Also I noticed a flaw code for non-bacterias, good you mentioned it here.